packages/broker/test/integration/plugins/operator/MaintainOperatorValueService.test.ts

```diff
@@ -0,0 +1,133 @@
+
+const logger = new Logger(module)
+
+const SPONSOR_AMOUNT = 200
+const STAKE_AMOUNT = 100
+
+describe("MaintainOperatorValueService", () => {
+    let provider: Provider
+    let operatorWallet: Wallet
+    let operatorContract: Operator
+    let token: IERC677
+    let adminWallet: Wallet
+    let streamId1: string
+    let streamId2: string
+
+    let operatorConfig: OperatorServiceConfig
+
+    const deployNewOperator = async () => {
+        const operatorWallet = await generateWalletWithGasAndTokens(provider)
+        logger.debug("Deploying operator contract")
+        const operatorContract = await deployOperatorContract(operatorWallet)
+        logger.debug(`Operator deployed at ${operatorContract.address}`)
+        operatorConfig = {
+            operatorContractAddress: operatorContract.address,
+            provider,
+            theGraphUrl,
+            fetch,
+            signer: operatorWallet
+        }
+        return { operatorWallet, operatorContract }
+    }
+
+    const getDiffBetweenApproxAndRealValues = async (): Promise<bigint> => {
+        const { sponsorshipAddresses, approxValues, realValues } = await operatorContract.getApproximatePoolValuesPerSponsorship()
+        let diff = BigInt(0)
+        for (let i = 0; i < sponsorshipAddresses.length; i++) {
+            diff = realValues[i].toBigInt() - approxValues[i].toBigInt()
+        }
+        return diff
+    }
+
+    beforeEach(async () => {
+        provider = getProvider()
+        logger.debug("Connected to: ", await provider.getNetwork())
+
+        adminWallet = new Wallet(ADMIN_WALLET_PK, provider)
+
+        token = new Contract(config.contracts.LINK, tokenABI, adminWallet) as unknown as IERC677
+        const client = new StreamrClient({
+            ...CONFIG_TEST,
+            auth: {
+                privateKey: ADMIN_WALLET_PK
+            }
+        })
+        streamId1 = (await client.createStream(`/operatorvalueservicetest-1-${Date.now()}`)).id
+        streamId2 = (await client.createStream(`/operatorvalueservicetest-2-${Date.now()}`)).id
+        await client.destroy();
+
+        ({ operatorWallet, operatorContract } = await deployNewOperator())
+
+        await (
+            await token.connect(operatorWallet).transferAndCall(operatorContract.address, parseEther(`${STAKE_AMOUNT * 2}`), operatorWallet.address)
+        ).wait()
+        for (const streamId of [streamId1, streamId2]) {
+            const sponsorship = await deploySponsorship(config, operatorWallet, { streamId })
+            await (await token.connect(operatorWallet).transferAndCall(sponsorship.address, parseEther(`${SPONSOR_AMOUNT}`), "0x")).wait()
+            await (await operatorContract.stake(sponsorship.address, parseEther(`${STAKE_AMOUNT}`))).wait()
+            expect(await token.balanceOf(sponsorship.address)).toEqual(parseEther(`${SPONSOR_AMOUNT + STAKE_AMOUNT}`))
```

Comment:
Why `expect`? Maybe should remove the expect assertion as this is just a test setup, not checking any logic about the service?

Reply:
Will remove since they're not testing this service, but rather the contract implementation